tethysext/atcore/services/resource_spatial_manager.py

```diff
@@ -49,25 +58,24 @@ def get_extent_for_project(self, datastore_name, resource_id):
 
         return extent
 
-    def get_projection_units(self, resource, srid):
-        pass
-
-    def get_projection_string(self, model_db, srid, proj_format=''):
-        pass
-
-    def create_extent_layer(self, datastore_name, resource_id):
+    def create_extent_layer(self, datastore_name, resource_id, geometry_type=None):
         """
         Creates a GeoServer SQLView Layer for the extent from the resource.
 
         Args:
             datastore_name(str): name of the app database, for example: app_primary_db.
-            resource_id(str): id of the extent record.
+            resource_id(str): id of the Resources.
+            geometry_type(str): type of geometry. Pick from: Polygon, LineString, Point.
         """
+        geometry_check_list = [x.lower() for x in [self.GT_POLYGON, self.GT_LINE, self.GT_POINT]]
+        if geometry_type is None or geometry_type.lower() not in geometry_check_list:
```

Comment:
Default to GT_POLYGON if no value is given. Raise a ValueError if an invalid value is given so the behavior isn't confusing.